### PR TITLE
Conditionally enable/disable targeting pack tests

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Addresses part of #14832.

### CODE REVIEW

File: src/Framework/test/TargetingPackTests.cs [hunk 1]
```diff
@@ -20,17 +20,24 @@ public class TargetingPackTests
...
 
         public TargetingPackTests(ITestOutputHelper output)
         {
             _output = output;
             _expectedRid = TestData.GetSharedFxRuntimeIdentifier();
             _targetingPackRoot = Path.Combine(TestData.GetTestDataValue("TargetingPackLayoutRoot"), "packs", "Microsoft.AspNetCore.App.Ref", TestData.GetTestDataValue("TargetingPackVersion"));
+            _isTargetingPackBuilding = bool.Parse(TestData.GetTestDataValue("IsTargetingPackBuilding"));
         }
 
         [Fact(Skip="https://github.com/aspnet/AspNetCore/issues/14832")]
```
Comment (wtgodbe) [inline src/Framework/test/TargetingPackTests.cs]: Should also unskip these 2

Comment (JunTaoLuo) [reply src/Framework/test/TargetingPackTests.cs]: Ah, I rebased from master, which didn't have these skipped.

**Review by wtgodbe — APPROVED** [review]

**Review by dougbu — APPROVED** [review]
Works for me though you should retry to make sure the tests are skipped correctly on the platform where the CI hit a 503.

**Comment by dougbu** [comment]
Please also submit a test (public) build with `IsTargetingPackBuilding` set to `true` e.g. duplicating whatever run you want and adding that variable. Should confirm tests do something -- perhaps w/ temporary logging after the `return`.

**Comment by JunTaoLuo** [comment]
Verified that it works as intended via file logging. I'll remove the logging code and merge.

**Pull request merged**